pkg/transformers/greenmask/greenmask_boolean_transformer.go

```diff
@@ -13,12 +13,24 @@ type BooleanTransformer struct {
 	transformer *greenmasktransformers.RandomBoolean
 }
 
-var booleanTransformerParams = []string{"generator"}
+var (
+	BooleanParams = []transformers.TransformerParameter{
```

Comment:
Rather than exposing all the transformer internals, maybe we can add a function that returns the transformer definition while keeping all these unexported? That function can then be called when building the map.

```
func BooleanTransformerDefinition() *transformers.Definition {
	return &transformers.Definition{
		SupportedTypes: booleanCompatibleTypes,
		Parameters:     booleanParams,
	}
}
```

This way the implementation details of the transformer are owned by the transformers package and can't be modified outside of it. Think of exported variables as global variables. Anything that uses this package has access to it and can modify it, which could result in unexpected behaviour down the line.